File: 238-productOfArrExceptSelf.js
function productArr(nums) {
    let prefix = 1; 
    let suffix = 1; 
    let res = [];
    for (let i = 0; i < nums.length; i++) {
        res.push(prefix);
        prefix *= nums[i];
    }
    for (let i = nums.length - 1; i >= 0; i--) {
        res[i] *= suffix;
        suffix *= nums[i];     
    }
    return res;
}

console.log(productArr(nums = [1,2,3,4]))
// Given an integer array nums, return an array answer such that answer[i] is equal to the product of all the elements of nums except nums[i].

// The product of any prefix or suffix of nums is guaranteed to fit in a 32-bit integer.

// You must write an algorithm that runs in O(n) time and without using the division operation.

function productList(nums) {
    
}

File: 238-productOfArrExceptSelf.py
def productExceptSelf(nums):
    prefix = 1
    suffix = 1
    res = []
    for i in range(len(nums)):
        res.append(prefix)
        prefix *= nums[i]
       
    i = len(nums) - 1
    while i >= 0:
        res[i] *= suffix
        suffix *= nums[i]
        i -= 1
    return res

print(productExceptSelf([1,2,3,4]))
